testing/aws-testing-cdk/lib/kinesis/KinesisSourceStack.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Stack, StackProps} from 'aws-cdk-lib';
+import {Construct} from 'constructs';
+import {Effect, Role, PolicyDocument, PolicyStatement, Policy} from 'aws-cdk-lib/aws-iam';
+
+export interface KinesisSourceStackProps extends StackProps {
+  readonly testingRole: Role;
+}
+
+/**
+ * CDK stack that creates a Kinesis stream
```

Comment:
Is this stack supposed to create a Kinesis stream or not? I only see IAM policies being created.

Reply:
The stream will be created as part of the test. The stack only adds the relevant permissions for accessing Kinesis streams and accessing the dynamoDb table for lease coordination.

Reply:
Will do a separate PR for adding the README.